test/Sign.Core.Test/Tools/VSIXSignTool/OpcPackageTests.cs

```diff
@@ -0,0 +1,214 @@
+        {
+            using (var package = OpcPackage.Open(SamplePackage))
+            {
+                var newItem = new OpcContentType("test", "test", OpcContentTypeMode.Default);
+                var contentTypes = package.ContentTypes;
+                Assert.Throws<InvalidOperationException>(() => contentTypes.Add(newItem));
+            }
+        }
+
+        [Fact]
+        public void ShouldAllowUpdatingContentType()
+        {
+            int initialCount;
+            string shadowPath;
+            using (var package = ShadowCopyPackage(SamplePackage, out shadowPath, OpcPackageFileMode.ReadWrite))
+            {
+                initialCount = package.ContentTypes.Count;
+                var newItem = new OpcContentType("test", "application/test", OpcContentTypeMode.Default);
+                package.ContentTypes.Add(newItem);
+            }
+            using (var reopenedPackage = OpcPackage.Open(shadowPath))
+            {
+                Assert.Equal(initialCount + 1, reopenedPackage.ContentTypes.Count);
+            }
+        }
+
+        [Fact]
+        public void ShouldAllowUpdatingRelationships()
+        {
+            int initialCount;
+            string shadowPath;
+            using (var package = ShadowCopyPackage(SamplePackage, out shadowPath, OpcPackageFileMode.ReadWrite))
+            {
+                initialCount = package.Relationships.Count;
+                var newItem = new OpcRelationship(new Uri("/test", UriKind.RelativeOrAbsolute), new Uri("/test", UriKind.RelativeOrAbsolute));
+                package.Relationships.Add(newItem);
+                Assert.True(newItem.Id != null && newItem.Id.Length == 9);
+            }
+            using (var reopenedPackage = OpcPackage.Open(shadowPath))
+            {
+                Assert.Equal(initialCount + 1, reopenedPackage.Relationships.Count);
+            }
+        }
+
+        [Fact]
+        public void ShouldRemovePart()
+        {
+            using (var package = ShadowCopyPackage(SamplePackage, out _, OpcPackageFileMode.ReadWrite))
+            {
+                var partToRemove = new Uri("/extension.vsixmanifest", UriKind.Relative);
+                var part = package.GetPart(partToRemove);
+                Assert.NotNull(part);
+                package.RemovePart(part);
+                Assert.Null(package.GetPart(partToRemove));
+            }
+        }
+
+
+        [Fact]
+        public void ShouldRemoveRelationshipsForRemovedPartWhereRelationshipIsMaterialized()
+        {
+            string path;
+            using (var package = ShadowCopyPackage(SamplePackage, out path, OpcPackageFileMode.ReadWrite))
+            {
+                var partToRemove = new Uri("/extension.vsixmanifest", UriKind.Relative);
+                var part = package.GetPart(partToRemove);
+                part!.Relationships.Add(new OpcRelationship(new Uri("/foo", UriKind.Relative), new Uri("http://foo.com", UriKind.Absolute)));
```

Comment:
Avoid `foo`.
In tests, use the [`.test` domain](https://wikipedia.org/wiki/.test).

```suggestion
                part!.Relationships.Add(new OpcRelationship(new Uri("/test", UriKind.Relative), new Uri("http://opc.test", UriKind.Absolute)));
```